modules/gapi/test/gapi_gpu_test_kernels.cpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+#include "test_precomp.hpp"
+
+#include <iomanip>
+#include "gapi_gpu_test_kernels.hpp"
+#include <opencv2/gapi/core.hpp>
+#include "opencl_kernels_test_gapi.hpp"
+
+
+namespace cv
+{
+#ifdef HAVE_OPENCL
+namespace gapi_test_kernels
+{
+
+GAPI_GPU_KERNEL(GGPUSymm7x7_test, TSymm7x7_test)
+{
+    static void run(const cv::UMat& in, cv::UMat &out)
+    {
+        if (cv::ocl::isOpenCLActivated())
+        {
+            cv::Size size = in.size();
+            size_t globalsize[2] = { (size_t)size.width, (size_t)size.height };
+
+            //size_t maxWorkItemSizes[32];
+            //cv::ocl::Device::getDefault().maxWorkItemSizes(maxWorkItemSizes);
+
+            cv::ocl::Kernel kernel;
+
+            int coefficients[10] = { 1140, -118, 526, 290, -236, 64, -128, -5, -87, -7 };
+            int shift = 10;
```

Comment:
So real kernels usually have this as parameters (of the kernel) - can it be made here so? Then it would be more "complete" example showing how almost arbitrary parameters can be passed from G-API to the underlying OpenCL kernel

Reply:
Done